source/usage-examples/bulkWrite.txt

```diff
@@ -0,0 +1,93 @@
+=======================
+Perform Bulk Operations
+=======================
+
+.. default-domain:: mongodb
+
+Overview
+--------
+
+:node-api:`collection.bulkWrite <Collection.html#bulkWrite>` lets you
+perform bulk write operations against a *single* collection. With
+``collection.bulkWrite()``, you specify a list of operations to perform and
+MongoDB then executes those operations in bulk. Bulk write supports
+``insertOne``, ``updateOne``, ``updateMany``, ``deleteOne``,
+``deleteMany``, and ``replaceOne`` operations. Refer to the method
+documentation for full details.
+
+``bulkWrite()`` accepts the following parameters:
+
+- ``operations``: specifies the bulk operations to
+  perform. Each operation is passed to ``bulkWrite()`` as an object in
+  an array.
+
+- ``options``: *optional* settings that affect the execution
+  of the operation, such as :manual:`write concern
+  </reference/write-concern>` and order.
+
+  By default, MongoDB executes bulk operations in the specified order
+  (i.e. serially). When executing operations in order, if an error occurs
+  during the processing MongoDB returns without processing any of the
+  remaining operations in the list. In contrast, when ``ordered`` is
+  ``false``, MongoDB continues to process remaining write operations in
+  the list. Unordered operations are faster as MongoDB can execute the
+  operations in parallel, but the results of the operation may vary. For
+  example, a ``deleteOne`` operation run before an ``updateMany``
+  operation might have a different result from running it after the
+  ``updateMany``. Refer to :ref:`Execution of Operations
+  <bulkwrite-write-operations-executionofoperations>` for more
+  information.
+
+- ``callback``: command result callback. Like other collection methods,
+  ``bulkWrite()`` returns a Promise if you do not specify a callback. The
+  Promise resolves to a :node-api:`bulkWriteOpCallback </Collection.html#~bulkWriteOpCallback>` object containing the
+  result of the operation.
+
+``bulkWrite()`` returns a ``BulkWriteResult`` object
+that includes the number of updated or inserted documents, the ``_id`` of
+each document, and any ``writeError`` or ``writeConcernErrors``. Refer to
+the documentation for :node-api:`BulkWriteResult </BulkWriteResult>` for
+more information.
+
+If you create an index with a :ref:`unique constraint
```

Comment:
the link here is broken, sadly. I also think "If you create an index" might give the wrong impression -- this is really talking about "If you perform bulk operations on a collection with a unique constraint", so that might be a clearer way of expressing this.

Reply:
(I think the ref link destination doesn't exist here?)

Reply:
Historically, we've been able to use refs from the manual in other projects. I'm not sure if that's something nextgen supports yet, so I'll just link to the manual page directly. I've also rephrased the paragraph.